### DISCUSSION TITLE
Can LaravelEventSourcing work with Doctrine?

### DISCUSSION THREAD

**Discussion by ncphillips** [spatie/laravel-event-sourcing#421] [Q&A] [answered]
Our app uses doctrine rather then Eloquent.

I'm wondering if LaravelEventSourcing will work with that setup?

**Comment by etahamer** [comment, marked as answer]
@ncphillips Read this one https://spatie.be/docs/laravel-event-sourcing/v7/advanced-usage/using-your-own-event-storage-repository

**Reply by sebastiandedeyne** [reply]
Our package is built to support any data store, so Doctrine should work fine as long as you implement a custom repository.

**Reply by ncphillips** [reply]
Thanks!

We wound up just using Eloquent. It's pretty self-contained so Idon't think we'd ever really interact with other entities so it should be fine.

Maybe someday we'll look to set it up, but it's not something we really want to have to maintain.
